MantidPlot/src/ProjectRecoveryGUIs/ProjectRecoveryPresenter.cpp

```diff
@@ -0,0 +1,190 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "ProjectRecoveryPresenter.h"
+#include "ApplicationWindow.h"
+#include "ProjectRecovery.h"
+#include "ProjectRecoveryModel.h"
+#include "ProjectRecoveryView.h"
+#include "RecoveryFailureView.h"
+
+#include <QDialog>
+#include <memory>
+
+ProjectRecoveryPresenter::ProjectRecoveryPresenter(
+    MantidQt::ProjectRecovery *projectRecovery, ApplicationWindow *parentWindow)
+    : m_mainWindow(parentWindow) {
+  m_recView = nullptr;
+  m_failureView = nullptr;
+  m_model = new ProjectRecoveryModel(projectRecovery, this);
+  m_openView = RecoveryView;
+  m_startMantidNormallyCalled = false;
```

Comment:
Why not initialize these fields in the initialization thing?

Reply:
I believe it was from an early implementation, should be sorted :+1: